var/spack/repos/builtin/packages/fish/package.py

```diff
@@ -41,6 +43,13 @@ def determine_version(cls, exe):
         match = re.search(r'fish, version (\S+)', output)
         return match.group(1) if match else None
 
+    def url_for_version(self, version):
+        if version < spack.version.Version('3.2.0'):
+            url = 'https://github.com/fish-shell/fish-shell/releases/download/{0}/fish-{0}.tar.gz'
+        else:
+            url = 'https://github.com/fish-shell/fish-shell/releases/download/{0}/fish-{0}.tar.xz'
+        return url.format(version)
```

Comment:
This could be a bit simpler:
```suggestion
        url = 'https://github.com/fish-shell/fish-shell/releases/download/{0}/fish-{0}.tar.{1}'
        if version < Version('3.2.0'):
            ext = 'gz'
        else:
            ext = 'xz'
        return url.format(version, ext)
```

Reply:
yep that's better thanks 👍